tests/TestCases/ValidatableTest.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+use Illuminate\Validation\ValidationException;
+
+use Tests\TestCases\TestCase;
+
+use Tests\Helpers\Models\MValidatableAttributes;
+use Tests\Helpers\Models\MValidatableFunctions;
+use Tests\Helpers\Models\MValidatableDataAccessPriority;
+
+class ValidatableTest extends TestCase
```

Comment:
I think we should have a test that the validation runs on `save()` as expected, which you've got! But I think we should also test that calling the `validate()` function manually, and that it returns the correct values, etc.

I notice a lot of `assertTrue(true)` which is useful on occasion, but a majority of the time that pattern is actually an indication that you're not approaching the test correctly. In other words, we should look for something explicit to assert. So for example, calling `validate` manually should return an array of the validated attributes, which is a value you assert your expectations for.

Reply:
Let's talk on Monday about what we can explicitly assert in these cases, because I've been in your shoes, i know it can be tricky